lib/openstack_api/openstack_hypervisor.py

```diff
@@ -0,0 +1,54 @@
+from enum import Enum, auto
+from typing import List
+
+from enums.query.props.hypervisor_properties import HypervisorProperties
+from enums.query.query_presets import QueryPresetsGeneric
+from openstack_query import HypervisorQuery
+
+
+class _HypervisorSearchTypes(Enum):
+    EMPTY_HVS = auto()
+
+
+# pylint: disable=too-few-public-methods
+class OpenstackHypervisor:
+    """
+    Class that contains methods for OpenStack hypervisors
+    """
+
+    def __init__(self, query_class=HypervisorQuery):
+        """
+        Initializes the class with an optional hook for DI
+        """
+        super().__init__()
+        self._query_cls = query_class
+
+    def find_hypervisor(self, cloud_account: str, search_type: str):
+        """
+        Runs a search for hypervisors based on the search type
+        """
+        try:
+            search_type = _HypervisorSearchTypes[search_type]
+        except KeyError as exc:
+            raise ValueError(f"Invalid search type: {search_type}") from exc
+
+        if search_type == _HypervisorSearchTypes.EMPTY_HVS:
+            return self._find_empty_hypervisors(cloud_account)
+
+        raise NotImplementedError(f"Search type {search_type} not implemented")
+
+    def _find_empty_hypervisors(self, cloud_account: str) -> List[str]:
```

Comment:
prefer this to be a public method - because then we can use it in other workflows